Installation/cmake/modules/CGAL_CreateSingleSourceCGALProgram.cmake

```diff
@@ -12,26 +33,45 @@ function(create_single_source_cgal_program firstfile )
 
     # remaining files
     foreach( i ${ARGN} )
-      set( all ${all} ${CMAKE_CURRENT_SOURCE_DIR}/${i} )
+      set( all ${all} ${CGAL_CURRENT_SOURCE_DIR}/${i} )
     endforeach()
 
 
     add_executable(${exe_name} ${all})
 
     if(BUILD_TESTING)
-      if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${exe_name}.cin")
-        set(ARGS "${CMAKE_CURRENT_SOURCE_DIR}/${exe_name}.cin")
-      elseif(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${exe_name}.cmd")
-        file(STRINGS "${CMAKE_CURRENT_SOURCE_DIR}/${exe_name}.cmd" ARGS LIMIT_COUNT 1)
-        # TODO: handle multi-lines .cmd files
-        # see https://github.com/CGAL/cgal/pull/1295/files/c65d3abe17bb3e677b8077996cdaf8672f9c4c6f#r71705451
+      set(cin_file "${CGAL_CURRENT_SOURCE_DIR}/${exe_name}.cin")
+      if(EXISTS ${cin_file})
+	add_test(NAME ${exe_name}
+          COMMAND ${CMAKE_COMMAND}
+	    -DCMD:STRING=$<TARGET_FILE:${exe_name}>
+	    -DCIN:STRING=${cin_file}
+	    -P "${CGAL_MODULES_DIR}/run_test_with_cin.cmake")
+#	message(STATUS "add test: ${exe_name} < ${cin_file}")
+      else()
+	# TODO: deal with shell globbing; if the `cmd` file contains
+	# a `*`, then interprete the command using bash.
+	set(cmd_file "${CGAL_CURRENT_SOURCE_DIR}/${exe_name}.cmd")
+	if(EXISTS ${cmd_file})
+          file(STRINGS "${cmd_file}" CMD_LINES)
+	  set(ARGS)
+#	  message(STATUS "DEBUG test ${exe_name}")
+	  foreach(CMD_LINE ${CMD_LINES})
+#	    message(STATUS "  command line: ${CMD_LINE}")
+	    separate_arguments(CMD_LINE_ARGS UNIX_COMMAND ${CMD_LINE})
+#	    message(STATUS "  args: ${CMD_LINE_ARGS}")
+	    list(APPEND ARGS ${CMD_LINE_ARGS})
+	  endforeach()
+	  expand_list_with_globbing(ARGS)
+	endif()
+#	message(STATUS "add test: ${exe_name} ${ARGS}")
+	add_test(NAME ${exe_name} COMMAND ${exe_name} ${ARGS})
```

Comment:
The current documentation of `.cmd` files says (see `Guidelines/Testing#.2A.cmd_files` in our wiki):
> For each line in program.cmd, the testsuite will then execute the program with the full line as arguments.

but that is not coherent with the current behavior.

If needed, the documented behavior is rather easy to implement.